internal/events/event.go

```diff
@@ -314,14 +317,30 @@ func (e Events) queryEvents(ctx context.Context, selectedCategories []string, cl
 	}
 
 	if result == nil { //non supported option
-		return make([]*common.Event, 0), nil
+		return make([]*common.Event, 0), swag.Int64(0), nil
 	}
 
 	var events []*common.Event
-	return events, WithIDs(result).Find(&events).Error
+
+	result = WithIDs(result)
+	totalRecords := result.Find(&events).RowsAffected
+
+	// if limit not specified or not valid, we set it to default. -1 is a valid value for limit which means no limit.
+	if limit == nil || *limit < -1 || *limit == 0 {
+		limit = swag.Int64(defaultPageSize)
+	}
+
+	// if offset not specified, we return the first page. If offset is specified but not valid, we return an error.
+	if offset == nil {
+		offset = swag.Int64(0)
+	} else if *offset < 0 || *offset > totalRecords {
```

Comment:
Why would we return error with offset larger than totalRecords? It would just return 0 results which I think it's acceptable (same behaviour as SQL). In any case this validation would be easier with the page-oriented approach - please consider validating at swagger level what we can.